idb/postgres/postgres.go

```diff
@@ -242,60 +240,85 @@ func (db *IndexerDb) AddBlock(block *bookkeeping.Block) error {
 			}
 		}
 
+		return nil
+	}
+	// We split functionality in two functions, `f` and `ff`, so that we can easily free
+	// the database resources once and only once using defer before committing the database
+	// transaction.
+	ff := func(tx pgx.Tx) error {
```

Comment:
Replace this with `return tx.Commit` in f?

Reply:
If we do, we would have the same problem as before - closing prepared statements after committing the transaction.

Reply:
Oh, I thought that was what you moved into TxRetry. Is there a way to fix this in the helper function?

Reply:
Fix what?

Reply:
The situation where using a helper function incorrectly leads to unexpected orphaned connections.

Reply:
Since we now call Rollback in `TxWithRetry`, leaving opened transactions is not an issue. There can be other issues like using the transaction after committing it. We could commit in `TxWithRetry`

Reply:
Ok. In that case, I think this can stay as-is, maybe we can have a follow-up that commits in TxWithRetry.